byron/ledger/executable-spec/src/Ledger/Update.hs

```diff
@@ -520,10 +529,14 @@ instance STS UPVOTE where
     , Set (UpId, Core.VKeyGenesis)
     )
   type Signal UPVOTE = Vote
+
+  -- | `S_CfmThdNotReached` is a structural `PredicateFailure`, used to fail
+  -- from one transition rule to the other. The other `PredicateFailure`s are
+  -- all throwable.
   data PredicateFailure UPVOTE
     = ADDVOTEFailure (PredicateFailure ADDVOTE)
     | HigherThanThdAndNotAlreadyConfirmed
-    | CfmThdNotReached
+    | S_CfmThdNotReached
     | AlreadyConfirmed
```

Comment:
This shows that it's not so easy to see which predicate failures are cannot be failures. `AlreadyConfirmed` cannot be thrown either.

Reply:
Ah right! I missed the double-negation.